pgjdbc/src/test/java/org/postgresql/test/jdbc2/NumericTransferTest.java

```diff
@@ -0,0 +1,81 @@
+package org.postgresql.test.jdbc2;
+
+import static org.junit.Assert.assertEquals;
+
+import org.postgresql.PGProperty;
+import org.postgresql.core.Oid;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Properties;
+
+@RunWith(Parameterized.class)
+public class NumericTransferTest extends BaseTest4 {
+  public NumericTransferTest(BinaryMode binaryMode) {
+    setBinaryMode(binaryMode);
+  }
+
+  @Override
+  protected void updateProperties(Properties props) {
+    super.updateProperties(props);
+    PGProperty.BINARY_TRANSFER_ENABLE.set(props, Oid.NUMERIC);
```

Comment:
is the effects of this limited to this test?

Reply:
In practice, it is not enough. 
https://github.com/pgjdbc/pgjdbc/pull/1637 makes it impossible to activate custom oids even with "I know what I am doing" mode.

I believe we should remove `binaryOids.retainAll(SUPPORTED_BINARY_OIDS);`

Reply:
> I believe we should remove binaryOids.retainAll(SUPPORTED_BINARY_OIDS);

I agree. It makes the ability to add oids by property completely useless. It is a property which should be rarely, and only very carefully, used, but I see no point in disabling it.

Reply:
I verified that removal of `retainAll(SUPPORTED_BINARY_OIDS)` works. In other words, the test in this PR fails without the fix.